_posts/2023-11-30-prevent-broker-scale-down-if-containing-paritition-replicas.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+title: "Preventing Kafka broker scale down if partition replicas are present"
+date: 2023-12-11
+author: Shubham Rawat
+---
+
+Apache Kafka is a platform designed for scalability.
+You can always scale in or scale out your Kafka Clusters based on your use case.
+When scaling down a cluster, it's crucial to ensure that data across brokers is moved or copied throughout the cluster.
+You may experience data loss if you forget to move out partition replicas from the broker being removed.
+But don't worry, with Strimzi 0.38, we have introduced the broker scale down check which is going to take care of this problem.
+
+## Broker scale down check
+
+This check makes sure that, when you are scaling down your Kafka cluster, there are no partition replicas present on the broker that is going to be removed.
+If partition replicas are found on the broker then the cluster operations are blocked and reconciliation fails until you revert the Kafka resource.
+This is enabled by default with Strimzi 0.38.
+
+However, there may be scenarios where you want to bypass this blocking mechanism.
+For example, disabling the check might be necessary when new topics are being constantly created.
+This situation can indefinitely block cluster operations, even when brokers are nearly empty.
+
+You can bypass the blocking mechanism by annotating the `Kafka` resource for the Kafka cluster by setting the `strimzi.io/skip-broker-scaledown-check` annotation to `true`:
+```shell
+kubectl annotate Kafka my-cluster strimzi.io/skip-broker-scaledown-check="true"
+```
+
+## Setting up the environment
+
+Let's set up a cluster to work through an example demonstrating this feature.
+
+To get the Kafka cluster up and running, we will first have to install the Strimzi Cluster Operator and then deploy the `Kafka` resource.
+You can refer to the [Stimzi Quickstart Guide](https://strimzi.io/docs/operators/latest/quickstart.html) to install the operator.
+
+You can install the Cluster Operator with any installation method you prefer.
+
+Then we will deploy the Kafka cluster with Cruise Control enabled.
```

Comment:
Would it make sense to join these into one paragraph? Not something you have to do. But I think the actual blog post looks better that way then with several parapgraphs consisiting from one sentence only.